### PR TITLE
Fix console error for userMenu Component

### PR DESCRIPTION
**Pull request by nikitharani**
Closes [TD-1776](https://sce.myjetbrains.com/youtrack/issue/TD-1776/Fix-console-errors-in-react-ui)

## Changes

- removed onchangePassword and onLogout props
- added UserAvatarProps

## Testing notes
on npm run test should not see any console errors for userMenu component

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/UserMenu/UserMenu.tsx [hunk 1]
```diff
@@ -95,7 +91,11 @@ export default function UserMenu({
  *
  * Provides custom styling and converts username to max 2 initials
  */
-const UserAvatar = ({ username, ...rest }: UserMenuProps) => {
+interface UserAvatarProps {
```
Comment (mattcorner) [inline src/UserMenu/UserMenu.tsx:94]: typescript props should live in UserMenu.types.ts even for internal sub components.

File: src/UserMenu/UserMenu.types.ts [hunk 2]
```diff
@@ -12,3 +12,7 @@ export interface UserMenuProps {
    */
   username: string;
 }
+
+export interface UserAvatarProps {
+  username: string;
```
Comment (m4manjesh) [inline src/UserMenu/UserMenu.types.ts:17]: missing comment

**Review by m4manjesh — CHANGES_REQUESTED** [review]
The changes look good to me. npm run test not throwing any errors. 

- [ ] Add the missing comment

**Review by m4manjesh — APPROVED** [review]
I will leave you to merge and release.

**Pull request merged**